enhancements/installer/openstack-no-fip-install.md

```diff
@@ -0,0 +1,99 @@
+---
+title: openstack-bring-your-own-external-connectivity
+authors:
+  - "@egarcia"
+  - "@adduarte"
+reviewers:
+  - "@mandre"
+  - "@pierre"
+approvers:
+  - "@mandre"
+  - "@pierre"
+creation-date: 2020-5-26
+last-updated: 2020-5-26
+status: implementable
+---
+
+# OpenStack BYO External Connectivity
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Numerous customers have requested the ability to run the OpenShift on OpenStack IPI installer without using Floating IP addresses.
+In the installer, we use floating IPs as a way to set up exeternal connectivity to the OpenShift cluster. Depending on how the OpenStack cluster
+is set up, the use of floating IPs may be impossible or very inconvenient. In order to support these customers, we intend to support installations
+on these types of OpenStack clusters.
+
+
+### Non-Goals
+
+This enhancement does not intend to provide any guidance or features that go further than allowing customers to deploy withoug Floating IPs. Operations like attaching loadbalancers to the entrypoints, or any day 2 operations are beyond the scope of this enhancement. 
+
+## Proposal
+
+In order to implement this feature fully, the following changes must be made:
+    1. Make `externalNetwork` an optional argument in the install-config
+    2. Disable `lbFloatingIP` when no external network is passed
+
+### User Stories
+
+#### Provider Networks
+
+As an enterprise OpenStack cluster administrator, I want to use the IPI installer but my cluster does not support Floating IPs. I need a way to install the OpenShift IPI installer on provider networks instead. I may also need to clear IP addresses I use on the provider network through an internal audit.
+
+##### Example Usage
+
+Using the proposed feature, the user would not set `externalNetwork` or `lbFloatingIP`. Then, they will set`machinesSubnet` to the ID of a provider network that they want to install the cluster onto. They will also set custom ports IPs for the `apiVP` and `ingressVIP` to ensure the IP they take is available. Here is an example install config:
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  openstack:
+    cloud: mycloud
+    computeFlavor: m1.s2.xlarge
+    externalNetwork:
+    lbFloatingIP:
+    machinesSubnet: fa806b2f-abcd-4bce-b9db-124bc64209bf
```

Comment:
Please state somewhere what the CIDR is on the `fa806b2f-abcd-4bce-b9db-124bc64209bf` subnet, to make it extra clear the `apiVIP` and `ingressVIP` addresses are picked from this range.

Reply:
Why dont I just add the machineNetwork parameter instead

Reply:
This could be validated automatically, right?  You could hit the neutron API to retrieve the details of this subnet.  Even if machineNetwork is specified, this validation would be helpful